PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -24,7 +28,14 @@ def __generate_seed(mnemonic: str, seed_password: str = '') -> bytes:
     :rtype: bytes
     :return: Seed
     """
-    pass
+    #who must check the encoding of both inputs? should be UTF-8 NFKD
+    #where should I catch UnicodeError?
+    mnemonic = mnemonic.encode()  #encoding string into bytes, UTF-8 by default
+    #in documentation, input is called a passphrase
+    #and "mnemonic"+passphrase is called a password. Do we need to change the names?
+    passphrase = "mnemonic" + seed_password
+    passphrase = passphrase.encode()
+    return pbkdf2_hmac('sha512', mnemonic, passphrase, PBKDF2_ROUNDS, SEED_LEN)
```

Comment:
>we use the PBKDF2 function with a mnemonic sentence (in UTF-8 NFKD) used as the password and the string "mnemonic" + passphrase (again in UTF-8 NFKD) used as the salt
> https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#from-mnemonic-to-seed

I think this is fine, PBKDF2 usually works with *password* and *salt*, therefore we should use mnemonic as the *password* for PBKDF2 and  `'mnemonic' + password_for_ protection_of_the_seed` as *salt*. It's quite confusing. :)